puppet/lib/puppet/util/puppetdb/char_encoding.rb

```diff
@@ -104,13 +86,14 @@ def self.error_char_context(str, bad_char_indexes)
   # @param error_context_str information about where this string came from for use in error messages
   # @return String
   def self.warn_if_invalid_chars(str, error_context_str)
-    bad_char_indexes = all_indexes_of_char(str, DEFAULT_INVALID_CHAR)
-    if bad_char_indexes.empty?
+
+    if str.index(DEFAULT_INVALID_CHAR).nil?
```

Comment:
This new code only checks the index of the first invalid character, just like https://github.com/puppetlabs/puppetdb/pull/1789/files#diff-a0488ae759c41269703997e51a97b388R102